### PR TITLE
[RTM] Automatic port of System.Net.Http.SocketsHttpHandler

### PR DESCRIPTION
**Pull request by carlossanlop**
These are the last APIs for 5.0, introduced in RC2: https://github.com/dotnet/runtime/pull/42563

@geoffkizer @aik-jahoda @scalablecory @gewarren @safern

### CODE REVIEW

File: xml/System.Net.Http/SocketsHttpConnectionContext.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
         <ReturnType>System.Net.Http.HttpRequestMessage</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>The initial HttpRequestMessage that is causing the connection to be created.</summary>
         <value>To be added.</value>
```
Comment (carlossanlop) [inline xml/System.Net.Http/SocketsHttpConnectionContext.xml:59]:
```suggestion
        <value>An HttpRequestMessage instance.</value>
```

File: xml/System.Net.Http/SocketsHttpHandler.xml [hunk 2]
```diff
@@ -205,7 +205,7 @@ If this change is undesirable, you can configure your application to use the old
         <ReturnType>System.Net.CookieContainer</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets or sets the managed cookie container object. </summary>
+        <summary>Gets or sets the managed cookie container object.?</summary>
```
Comment (carlossanlop) [inline xml/System.Net.Http/SocketsHttpHandler.xml:208]:
```suggestion
        <summary>Gets or sets the managed cookie container object.</summary>
```

File: xml/System.Net.Http/SocketsHttpHandler.xml [hunk 3]
```diff
@@ -571,7 +571,7 @@ For example, if the value is 64, then 65,536 bytes are allowed for the maximum r
         <ReturnType>System.Func&lt;System.Net.Http.SocketsHttpPlaintextStreamFilterContext,System.Threading.CancellationToken,System.Threading.Tasks.ValueTask&lt;System.IO.Stream&gt;&gt;</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Gets or sets a custom callback that provides access to the plaintext HTTP protocol stream.</summary>
         <value>To be added.</value>
```
Comment (carlossanlop) [inline xml/System.Net.Http/SocketsHttpHandler.xml:575]:
```suggestion
        <value>A callback that provides access to the plaintext HTTP protocol stream.</value>
```

File: xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml [hunk 4]
```diff
@@ -34,7 +34,7 @@
         <ReturnType>System.Net.Http.HttpRequestMessage</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>The initial HttpRequestMessage that is causing the stream to be used.</summary>
         <value>To be added.</value>
```
Comment (carlossanlop) [inline xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml:38]:
```suggestion
        <value>An HttpRequestMessage causing the stream to be used.</value>
```

File: xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml [hunk 5]
```diff
@@ -55,7 +55,7 @@
         <ReturnType>System.Version</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>The version of HTTP in use for this stream.</summary>
         <value>To be added.</value>
```
Comment (carlossanlop) [inline xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml:59]:
```suggestion
        <value>A version instance.</value>
```

File: xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml [hunk 6]
```diff
@@ -76,7 +76,7 @@
         <ReturnType>System.IO.Stream</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>The plaintext Stream that will be used for HTTP protocol requests and responses.</summary>
         <value>To be added.</value>
```
Comment (carlossanlop) [inline xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml:80]:
```suggestion
        <value>A stream instance.</value>
```

File: xml/System.Net.Http/SocketsHttpConnectionContext.xml [hunk 7]
```diff
@@ -55,8 +55,8 @@
         <ReturnType>System.Net.Http.HttpRequestMessage</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>The initial HttpRequestMessage that is causing the connection to be created.</summary>
+        <value>An HttpRequestMessage instance causing the connection to be created.</value>
         <remarks>To be added.</remarks>
```
Comment (safern) [inline xml/System.Net.Http/SocketsHttpConnectionContext.xml:60]: Why are the remarks empty?

Comment (carlossanlop) [reply xml/System.Net.Http/SocketsHttpConnectionContext.xml:60]: There were [no remarks in triple slash to port](https://github.com/dotnet/runtime/pull/42563/files), and the default string for empty remarks is `To be added. `

File: xml/System.Net.Http/SocketsHttpHandler.xml [hunk 8]
```diff
@@ -699,7 +699,7 @@ For example, if the value is 64, then 65,536 bytes are allowed for the maximum r
         <ReturnType>System.Net.IWebProxy</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets or sets the custom proxy when the <see cref="P:System.Net.Http.SocketsHttpHandler.UseProxy" /> property is <see langword="true" />.</summary>
+        <summary>Gets or sets the custom proxy when the?<see cref="P:System.Net.Http.SocketsHttpHandler.UseProxy" />?property is <see langword="true" />.</summary>
```
Comment (aik-jahoda) [inline xml/System.Net.Http/SocketsHttpHandler.xml:702]: What does the question marks mean?

File: xml/System.Net.Http/SocketsHttpConnectionContext.xml [hunk 9]
```diff
@@ -55,8 +55,8 @@
         <ReturnType>System.Net.Http.HttpRequestMessage</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>The initial HttpRequestMessage that is causing the connection to be created.</summary>
```
Comment (gewarren) [inline xml/System.Net.Http/SocketsHttpConnectionContext.xml:58]:
```suggestion
        <summary>Gets the initial HTTP request message that is causing the connection to be created.</summary>
```

File: xml/System.Net.Http/SocketsHttpConnectionContext.xml [hunk 10]
```diff
@@ -55,8 +55,8 @@
...
+        <value>An HttpRequestMessage instance causing the connection to be created.</value>
```
Comment (gewarren) [inline xml/System.Net.Http/SocketsHttpConnectionContext.xml:59]:
```suggestion
        <value>The request message that's causing the connection to be created.</value>
```

The actual type is shown automatically above the description. So we're not supposed to include it.

File: xml/System.Net.Http/SocketsHttpHandler.xml (hunk 8 above)
Comment (gewarren) [reply xml/System.Net.Http/SocketsHttpHandler.xml:702]: This change should be reverted.

File: xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml [hunk 11]
```diff
@@ -34,8 +34,8 @@
         <ReturnType>System.Net.Http.HttpRequestMessage</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>The initial HttpRequestMessage that is causing the stream to be used.</summary>
```
Comment (gewarren) [inline xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml:37]:
```suggestion
        <summary>Gets the initial HTTP request message that is causing the stream to be used.</summary>
```

File: xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml [hunk 12]
```diff
@@ -34,8 +34,8 @@
...
+        <value>An HttpRequestMessage causing the stream to be used.</value>
```
Comment (gewarren) [inline xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml:38]:
```suggestion
        <value>The HTTP request message that is causing the stream to be used.</value>
```

File: xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml [hunk 13]
```diff
@@ -55,8 +55,8 @@
         <ReturnType>System.Version</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>The version of HTTP in use for this stream.</summary>
```
Comment (gewarren) [inline xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml:58]:
```suggestion
        <summary>Gets the version of HTTP in use for this stream.</summary>
```

File: xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml [hunk 14]
```diff
@@ -76,8 +76,8 @@
         <ReturnType>System.IO.Stream</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>The plaintext Stream that will be used for HTTP protocol requests and responses.</summary>
```
Comment (gewarren) [inline xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml:79]:
```suggestion
        <summary>Gets the plaintext stream that will be used for HTTP protocol requests and responses.</summary>
```

File: xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml [hunk 15]
```diff
@@ -76,8 +76,8 @@
...
+        <value>A stream instance.</value>
```
Comment (gewarren) [inline xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml:80]:
```suggestion
        <value>The plaintext stream that will be used for HTTP protocol requests and responses.</value>
```

File: xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml [hunk 16]
```diff
@@ -55,8 +55,8 @@
         <ReturnType>System.Version</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>The version of HTTP in use for this stream.</summary>
+        <value>A version instance.</value>
```
Comment (gewarren) [inline xml/System.Net.Http/SocketsHttpPlaintextStreamFilterContext.xml:59]:
```suggestion
        <value>The version of HTTP in use for this stream.</value>
```

**Review by gewarren — APPROVED** [review]

File: xml/System.Net.Http/SocketsHttpHandler.xml (hunk 8 above)
Comment (carlossanlop) [reply xml/System.Net.Http/SocketsHttpHandler.xml:702]: Thanks for noticing. I noticed another one which I deleted but didn't catch this one. I have to check if the tool has a bug.

Comment (carlossanlop) [reply xml/System.Net.Http/SocketsHttpHandler.xml:702]:
```suggestion
        <summary>Gets or sets the custom proxy when the?<see cref="P:System.Net.Http.SocketsHttpHandler.UseProxy" /> property is <see langword="true" />.</summary>
```

**Comment by geoffkizer** [comment]
Thanks @carlossanlop -- looks good to me.

**Pull request merged**